volatility3/framework/layers/intel.py

```diff
@@ -338,33 +371,19 @@ def _page_is_valid(entry: int) -> bool:
         return bool((entry & 1) or ((entry & 1 << 11) and not entry & 1 << 10))
 
     def _handle_page_fault(self, layer_name, offset, invalid_bits, entry, description):
+        kernel = self._get_kernel_module()
+        if kernel is None:
+            raise exceptions.PagedInvalidAddressException(self.name, offset, invalid_bits, entry, "kernel module not found!")
+
         tbit = bool(entry & (1 << 11))
         pbit = bool(entry & (1 << 10))
-        unknown_bit = bool(entry & (1 << 7))
-        n = (entry >> 1) & 0xF
         vbit = bool(entry & 1)
-        # Handle transition page
-        if tbit and not pbit:
-            position = invalid_bits - 1
-            name, size, _ = self._find_structure_index(position)
-            index = self._mask(offset, position, position + size) >> (position + size)
-
-            # Grab the base address of the table we'll be getting the next entry from
-            base_address = self._mask(entry, self._maxphyaddr - 1, size + self._index_shift)
-
-            table = self._get_valid_table(base_address)
-            if table is None:
-                raise exceptions.PagedInvalidAddressException(self.name, offset, position + 1, entry,
-                                                              "Page Fault at entry " + hex(
-                                                                  entry) + " in table " + name)
-
-            # Read the data for the next entry
-            entry_data = table[(index << self._index_shift):(index << self._index_shift) + self._entry_size]
-            super()._translate_entry(offset, invalid_bits, entry)
+        entry &= ~self._get_invalid_pte_mask(kernel)
```

Comment:
I get a little bit touchy about `~` in python when dealing with values that might be different lengths.  I think xor would be equivalent to `& ~` and would be unambiguous for any length of value.

Reply:
Replaced it with a xor